### PR TITLE
test: add mysql-schema sample

### PR DESCRIPTION
**Pull request by scrocquesel**
- [x] refactor schema service to mock it in the test (I don't know how to have a working database)
- [x] use quarkus config object
- [x] as far I know, org.apache.commons.lang3.RandomStringUtils is not native compliant but I don't know of a replacement
- [x] README is not up to date

### CODE REVIEW

**Comment by scrocquesel** [comment]
#191

File: samples/mysql-schema/src/main/java/io/javaoperatorsdk/operator/sample/MySQLSchemaReconciler.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    @Override
+    public UpdateControl<MySQLSchema> reconcile(MySQLSchema schema, Context context) {
+        Log.infof("Reconciling MySQLSchema with name: %s", schema.getMetadata().getName());
+        var dbSchema = context.getSecondaryResource(Schema.class);
+        Log.debugf("Schema: %s found for: %s ", dbSchema, schema.getMetadata().getName());
+        try (Connection connection = schemaService.getConnection()) {
+            if (dbSchema.isEmpty()) {
+                Log.debugf("Creating Schema and related resources for: %s", schema.getMetadata().getName());
+                var schemaName = schema.getMetadata().getName();
+                String password = RandomStringUtils.random(16);
```
Comment (scrocquesel) [inline samples/mysql-schema/src/main/java/io/javaoperatorsdk/operator/sample/MySQLSchemaReconciler.java:55]: Doesn't not pass native compilation

**Comment by scrocquesel** [comment]
It seems we don't have the same indentation rules for pom.xml files. I'm using default from vscode. I think it's the formatter coming with `redhat.vscode-xml`.

**Comment by metacosm** [comment]
I'm using the eclipse formatted with Quarkus' rules… that should match the reformatting that you get when you build.

**Comment by scrocquesel** [comment]
> I'm using the eclipse formatted with Quarkus' rules… that should match the reformatting that you get when you build.

works for .java only I guess

**Comment by scrocquesel** [comment]
Superseded by #221

**Pull request closed without merging**